cpp/tests/CMakeLists.txt

```diff
@@ -306,6 +306,7 @@ ConfigureTest(
   iterator/optional_iterator_test_chrono.cu
   iterator/optional_iterator_test_numeric.cu
   iterator/indexalator_test.cu
+  iterator/sizes_to_offsets_iterator_test.cu
```

Comment:
It seems like most categories of tests are sorted by filename, except for iterator tests. We could fix that in this PR if you like.